mysql-test/suite/rocksdb/t/prefix_extractor_override.test

```diff
@@ -0,0 +1,103 @@
+while ($i <= 100) {
+  let $insert = INSERT INTO t1 VALUES(1, $i, $i, $i);
+  eval $insert;
+  inc $i;
+}
+--enable_query_log
+set global rocksdb_force_flush_memtable_now = 1;
+
+--echo
+--echo Original Prefix Extractor:
+--echo
+--sorted_result
+SELECT * FROM information_schema.rocksdb_cf_options WHERE option_type like '%prefix_extractor%';
+
+# BF used (4+8+8+8)
+select variable_value into @u from performance_schema.global_status where variable_name='rocksdb_bloom_filter_prefix_checked';
+SELECT COUNT(*) FROM t1 WHERE id1=1 AND id2=1 AND id3=1;
+--let $assert_text="Expected exactly one bloom filter prefix check"
+--let $assert_cond="[select variable_value-@u from performance_schema.global_status where variable_name=\'rocksdb_bloom_filter_prefix_checked\']" = 1
+--source include/assert.inc
+
+--echo
+--echo Prefix Extractor (after override_cf_options set, should not be changed):
+--echo
+--sorted_result
+SELECT * FROM information_schema.rocksdb_cf_options WHERE option_type like '%prefix_extractor%';
+
+# This should no longer crash. See https://github.com/facebook/mysql-5.6/issues/641
+--echo
+--echo Restarting with new Prefix Extractor...
+--echo
+--let $restart_parameters=restart:--loose-rocksdb_override_cf_options=cf1={prefix_extractor=capped:26};
+--source include/restart_mysqld.inc
+
+--echo
+--echo Changed Prefix Extractor (after restart):
+--echo
+--sorted_result
+SELECT * FROM information_schema.rocksdb_cf_options WHERE option_type like '%prefix_extractor%';
+
+# Satisfies can_use_bloom_filter (4+8+8+8), but can't use because the old SST
+# files have old prefix extractor
+select variable_value into @u from performance_schema.global_status where variable_name='rocksdb_bloom_filter_prefix_checked';
+SELECT COUNT(*) FROM t1 WHERE id1=1 AND id2=1 AND id3=1;
+--let $assert_text="Expected no bloom filter prefix check"
+--let $assert_cond="[select variable_value-@u from performance_schema.global_status where variable_name=\'rocksdb_bloom_filter_prefix_checked\']" = 0
+--source include/assert.inc
+
+# Insert more data into t1, verify it uses new bloom filter
+--disable_query_log
+let $i = 101;
+while ($i <= 200) {
+  let $insert = INSERT INTO t1 VALUES(1, $i, $i, $i);
+  eval $insert;
+  inc $i;
+}
+--enable_query_log
+
+set global rocksdb_force_flush_memtable_now = 1;
+
+# BF used w/ new prefix extractor (4+8+8+8) (still increments once because it
+# needs to check the new SST file, but doesnt increment for SST file with old
+# extractor)
+select variable_value into @u from performance_schema.global_status where variable_name='rocksdb_bloom_filter_prefix_checked';
+SELECT COUNT(*) FROM t1 WHERE id1=1 AND id2=1 AND id3=1;
+--let $assert_text="Expected exactly one bloom filter prefix check"
+--let $assert_cond="[select variable_value-@u from performance_schema.global_status where variable_name=\'rocksdb_bloom_filter_prefix_checked\']" = 1
+--source include/assert.inc
+
+# should have 2 sst files, one with old prefix extractor and one with new
```

Comment:
Nit: comment "should have 2 sst files" != code "SELECT COUNT(*) != 1"

Reply:
Yeah, there was actually an upstream discussion on this test when first introduced that there are several assumptions made in the test that are not really guaranteed to be true forever. They agreed to 'let it go' and see if instability problems arose. This was one of those assumptions or loose logic. The fully proper way to check this would be to use the file map and enumerate all files before and after and ensure the diff is what is expected.